roadmap/implementers-guide/src/node/collators/collation-generation.md

```diff
@@ -1,9 +1,36 @@
 # Collation Generation
 
-> TODO
+The collation generation subsystem is executed on collator nodes and produces candidates to be distributed to validators. If configured to produce collations for a para, it produces collations and then feeds them to the [Collator Protocol][CP] subsystem, which handles the networking.
 
 ## Protocol
 
+Input: None
+
+Output: CollationDistributionMessage
+
 ## Functionality
 
-## Jobs, if any
+The process of generating a collation for a parachain is very parachain-specific. As such, the details of how to do so are left beyond the scope of this description. The subsystem should be implemented as an abstract wrapper, which is aware of this configuration:
+
+```rust
+struct CollationGenerationConfig {
+	key: CollatorPair,
+	collation_producer: Fn(params) -> async (HeadData, Vec<UpwardMessage>, PoV),
+}
+```
```

Comment:
Are you intending that collator nodes extend the polkadot client & recompile? It might be easier to have a local parachain-specific process talk to a polkadot node acting as a collator via some interprocess API, but that's a discussion for much later.

Reply:
> Are you intending that collator nodes extend the polkadot client & recompile

Basically, yes. That's how the current Cumulus architecture works, at least. Some IPC-based split would also be amenable to this approach, with the `collation_producer` yielding an IPC future.